src/gradata/hooks/duplicate_guard.py

```
@@ -0,0 +1,126 @@
+        if not watched_dir.exists():
+            continue
+        try:
+            for f in watched_dir.rglob("*.py"):
+                if f.name.startswith("__"):
+                    continue
+                sim = _similarity(target_path, f.name)
+                if sim > SIMILARITY_THRESHOLD:
+                    rel = str(f.relative_to(root))
+                    similar.append((rel, sim))
+        except Exception:
+            continue
```

Ruff (0.15.9)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)